docs/python/how-to-guides/sysadmin/setuptools-deployment.md

```diff
@@ -0,0 +1,407 @@
+
+if __name__ == "__main__":
+    app()
+```
+
+## Create pyproject.toml
+
+The `pyproject.toml` file is the modern standard for Python package configuration. It replaces `setup.py` and defines your package metadata, dependencies, and command-line scripts.
+
+Here's an example `pyproject.toml`:
+
+```toml
+[build-system]
+requires = ["setuptools>=61.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "my-dh-tool"
+version = "0.1.0"
+description = "A Deephaven-based command-line tool"
+readme = "README.md"
+requires-python = ">=3.8"
+authors = [
+  { name = "Your Name", email = "your.email@example.com" },
+]
+classifiers = [
+  "Development Status :: 3 - Alpha",
+  "Intended Audience :: Developers",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+  "deephaven-server>=0.35.0",
+]
+
+[project.scripts]
+my-dh-tool = "my_dh_tool.cli:app"
+
+[tool.setuptools.packages.find]
+where = ["src"]
+```
+
+### Script configuration explained
+
+The key configuration for command-line scripts is in the `[project.scripts]` section:
+
+```toml
+[project.scripts]
+my-dh-tool = "my_dh_tool.cli:app"
+```
+
+This tells the build system to:
+
+- Create a command-line script named `my-dh-tool`.
+- When executed, call the `app()` function from the `cli` module in the `my_dh_tool` package.
+
+The format is: `command-name = "package.module:function"`
+
+## Build the package
+
+Once your package structure is complete, build a wheel archive that can be distributed and installed.
+
+First, install the build tool:
+
+```shell
+pip install build
+```
+
+Then build your package:
+
+```shell
+cd my_dh_tool
+python -m build
+```
+
+This creates two files in the `dist/` directory:
+
+- A source distribution (`.tar.gz`)
+- A wheel distribution (`.whl`)
+
+The wheel file can be distributed and installed on other systems.
+
+## Install the package
+
+### For development
+
+Install your package in editable mode for development:
+
+```shell
+pip install -e .
+```
+
+This allows you to make changes to your code without reinstalling.
+
+### For regular use
+
+Install from the built wheel:
+
+```shell
+pip install dist/my_dh_tool-0.1.0-py3-none-any.whl
+```
+
+### Using pipx (recommended for command-line tools)
+
+[pipx](https://pipx.pypa.io/) is the recommended way to install Python command-line tools, as it installs them in isolated environments:
+
+```shell
+pipx install .
+```
+
+Or from the wheel:
+
+```shell
+pipx install dist/my_dh_tool-0.1.0-py3-none-any.whl
+```
+
+After installation, the `my-dh-tool` command is available system-wide (or within your virtual environment).
+
+### Run without installing
+
+You can also run your tool directly from the source directory:
+
+```shell
+python -m my_dh_tool input_data.csv
+```
```

Comment:
This highlights some of the awkwardness of this document.  We are documenting running in two different ways:
1. run a package directly (calling `__main__.py`)
2. run a script created in `pyproject.toml`.

I don't think the idea of these two cases is presented well.  The two ideas blend together.  In some places we only illustrate running one way vs another.  When we do run something, we aren't clear which path we are running.  etc.